airguardian-core/src/main/kotlin/systems/ajax/motrechko/airguardian/repository/DroneRedisRepository.kt

```diff
@@ -0,0 +1,75 @@
+package systems.ajax.motrechko.airguardian.repository
+
+import com.mongodb.client.result.DeleteResult
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.data.redis.core.ReactiveRedisTemplate
+import org.springframework.data.redis.core.ScanOptions
+import org.springframework.stereotype.Repository
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.switchIfEmpty
+import reactor.kotlin.core.publisher.switchIfEmptyDeferred
+import systems.ajax.motrechko.airguardian.model.Drone
+import java.time.Duration
+
+@Repository
+class DroneRedisRepository(
+    private val droneRepository: DroneRepository,
+    private val reactiveRedisRepository: ReactiveRedisTemplate<String, Drone>,
+    @Value("\${spring.data.redis.key.prefix}") private val dronePrefix: String,
+    @Value("\${spring.data.redis.ttl.minutes}") private val ttlMinutes: String
+) : DroneCacheableRepository {
+
+    override fun save(entity: Drone): Mono<Drone> =
+        droneRepository.save(entity)
+            .flatMap { savedDrone ->
+                reactiveRedisRepository.opsForValue()
+                    .set(
+                        dronePrefix + savedDrone.id.toHexString(),
+                        savedDrone,
+                        Duration.ofMinutes(ttlMinutes.toLong())
+                    )
+                    .thenReturn(savedDrone)
```

Comment:
```suggestion
                saveDroneToRedisCache(savedDrone)
```
?

Reply:
Note! I've written the [same comment](https://github.com/VolodymyrM27/AirGuardian/pull/13#discussion_r1382641555) earlier

Reply:
I didn't understand the first time, my bad